File: Baekjoon/트리/1197.py
# https://www.acmicpc.net/problem/1197

v, e = map(int, input().split())
arr = [list(map(int, input().split())) for _ in range(e)]
arr.sort(key = lambda x : x[2])
parent = list(range(v + 1))
sum = 0


def union(a, b):
    a = find(a)
    b = find(b)

    if b < a:
        parent[a] = b
    else:
        parent[b] = a


def find(a):
    if a == parent[a]:
        return a
    parent[a] = find(parent[a])

    return parent[a]


for a, b, c in arr:
    if find(a) != find(b):
        union(a, b)
        sum += c

print(sum)

File: Baekjoon/트리/1197.cpp
// https://www.acmicpc.net/problem/1197

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int V, E;
int parent[10001];
vector<pair<int, pair<int, int>>> edges;

int find(int x) {
    if (x == parent[x]) {
        return x;
    }
    return parent[x] = find(parent[x]);
}

void merge(int x, int y) {
    x = find(x);
    y = find(y);
    if (x != y) {
        parent[x] = y;
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    cin >> V >> E;
    for (int i = 1; i <= V; i++) {
        parent[i] = i;
    }

    for (int i = 0; i < E; i++) {
        int a, b, c;
        cin >> a >> b >> c;
        edges.push_back({c, {a, b}});
    }

    sort(edges.begin(), edges.end());

    int ans = 0;
    for (int i = 0; i < E; i++) {
        int a = edges[i].second.first;
        int b = edges[i].second.second;
        if (find(a) != find(b)) {
            merge(a, b);
            ans += edges[i].first;
        }
    }

    cout << ans << '\n';
    return 0;
}